### PR TITLE
[yasm/yasm-tool] Incorporate yasm-tool into yasm

### LINKED ISSUE #18746: [yasm] Dependent library of yasm binary is missing
**Host Environment**
- OS: Windows 8.1 x64, up-to-date
- Compiler: MSVC 19.29.30038.1

**To Reproduce**
vcpkg install mpg123:x86-windows --> fails
vcpkg install mpg123:x64-windows --> fails
vcpkg install mpg123:x86-windows-static --> succeeds
vcpkg install mpg123:x64-windows-static --> succeeds

**Failure logs** 
**x86**
```
The following packages will be built and installed:
    mpg123[core]:x86-windows -> 1.28.0
Detecting compiler hash for triplet x86-windows...
Could not locate cached archive: C:\Users\Daniel\AppData\Local\vcpkg\archives\7f\7ff2d3bab8cfe01abcb3de9fd52e93f808c52c7bc03a32808457f3d82d7cc346.zip
Starting package 1/1: mpg123:x86-windows
Building package mpg123[core]:x86-windows...
-- Downloading https://sourceforge.net/projects/mpg123/files/mpg123/1.28.0/mpg123-1.28.0.tar.bz2/download;https://sourceforge.net/projects/mpg123/files/mpg123
et;https://sourceforge.net/projects/mpg123/files/mpg123/1.28.0/mpg123-1.28.0.tar.bz2/download?use_mirror=freefr;https://sourceforge.net/projects/mpg123/files/
s://sourceforge.net/projects/mpg123/files/mpg123/1.28.0/mpg123-1.28.0.tar.bz2/download?use_mirror=netix;https://sourceforge.net/projects/mpg123/files/mpg123/1
-- Extracting source D:/Code/Repos/vcpkg/downloads/mpg123-1.28.0.tar.bz2
-- Applying patch 0001-fix-checkcpuarch-path.patch
-- Using source at D:/Code/Repos/vcpkg/buildtrees/mpg123/src/4e333ee4f3-f9ae7d9a6e.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /m
    Working Directory: D:/Code/Repos/vcpkg/buildtrees/mpg123/x86-windows-dbg
    See logs for more information:
      D:\Code\Repos\vcpkg\buildtrees\mpg123\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:104 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:44 (vcpkg_build_cmake)
  ports/mpg123/portfile.cmake:33 (vcpkg_install_cmake)
  scripts/ports.cmake:139 (include)


Error: Building package mpg123:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: mpg123:x86-windows
  Vcpkg version: 2021-05-05-e8977e69d9a3fb462d9ad42013d83a7682706659
```
**x64**
```
D:\Code\Repos\vcpkg>vcpkg install mpg123:x64-windows
Computing installation plan...
The following packages will be built and installed:
    mpg123[core]:x64-windows -> 1.28.0
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\Daniel\AppData\Local\vcpkg\archives\7e\7ef4e1a094c74c70480583abe8b85b6aabc42d923dc1a5d95c544d2fe4667652.zip
Starting package 1/1: mpg123:x64-windows
Building package mpg123[core]:x64-windows...
-- Using D:/Code/Repos/vcpkg/downloads/mpg123-1.28.0.tar.bz2
-- Cleaning sources at D:/Code/Repos/vcpkg/buildtrees/mpg123/src/4e333ee4f3-f9ae7d9a6e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/Code/Repos/vcpkg/downloads/mpg123-1.28.0.tar.bz2
-- Applying patch 0001-fix-checkcpuarch-path.patch
-- Using source at D:/Code/Repos/vcpkg/buildtrees/mpg123/src/4e333ee4f3-f9ae7d9a6e.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /m
    Working Directory: D:/Code/Repos/vcpkg/buildtrees/mpg123/x64-windows-dbg
    See logs for more information:
      D:\Code\Repos\vcpkg\buildtrees\mpg123\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:104 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:44 (vcpkg_build_cmake)
  ports/mpg123/portfile.cmake:33 (vcpkg_install_cmake)
  scripts/ports.cmake:139 (include)


Error: Building package mpg123:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: mpg123:x64-windows
  Vcpkg version: 2021-05-05-e8977e69d9a3fb462d9ad42013d83a7682706659
```

**Additional context**
The issue was introduced in commit 5edbbf1a70a0a502e6362f51bf875b96469ed807, which updated the mpg123 version from 1.26.5  to 1.28.0. This obviously means that both the mpg123 source code and multiple port-related vcpkg files got changed, and I am not 100% if we have an actual vcpkg port bug here or if the src itself is the culprit. I mention this as I tried to manually compile the src using latest VS 2019, and the process terminated with exactly the same - long - error message you'll find in one of the logs attached. 

Very interesting is the fact that building static libs works perfectly fine, so I the issue has to be connected to the building of _dynamic_ libraries somehow.

Ah, before you ask, yes, I tested with my "external" Ninja and CMake installations removed from %PATH% and Program Files as well, with exactly the same results.

Attached you'll find all logs generated during the operations mentioned above.

[mpg123_vcpkg_logs.zip](https://github.com/microsoft/vcpkg/files/6745472/mpg123_vcpkg_logs.zip)

### PR DESCRIPTION
**Pull request by JackBoosY**
Since port yasm and yasm-tool repo is same and generate the same files, remove port yasm-tool and switch the related dependency item yasm-tool to yasm.

Fixes https://github.com/microsoft/vcpkg/issues/18746.

Replace #18816.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@BillyONeal I removed the script changes, please review again.

Thanks.

**Review by LilyWangLL — APPROVED** [review]

File: ports/yasm/portfile.cmake [hunk 1]
```diff
@@ -29,11 +29,11 @@ vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 
 if (BUILD_TOOLS)
+    vcpkg_copy_tools(TOOL_NAMES vsyasm yasm ytasm AUTO_CLEAN)
```
Comment (BillyONeal) [inline ports/yasm/portfile.cmake:32]: Won't `AUTO_CLEAN` delete `yasmstd${VCPKG_TARGET_SHARED_LIBRARY_SUFFIX}` that line 34 is trying to copy?

File: ports/yasm-tool/vcpkg.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
   "name": "yasm-tool",
-  "version": "1.3.0",
-  "port-version": 1,
-  "description": "A modular assembler.",
-  "homepage": "http://yasm.tortall.net/",
-  "supports": "native & !uwp"
+  "version-date": "2021-12-14",
+  "description": "Legacy port. Use port yasm instead.",
+  "dependencies": [
+    "yasm"
```
Comment (BillyONeal) [inline ports/yasm-tool/vcpkg.json:6]: Shouldn't this be `yasm[tools]` as done in `yasm-tool-helper`?

File: ports/gmp/vcpkg.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
       "platform": "windows"
     },
     {
-      "name": "yasm-tool",
+      "name": "yasm",
       "host": true,
       "platform": "windows"
```
Comment (BillyONeal) [inline ports/gmp/vcpkg.json:17]: I think this needs the "tools" feature?

File: ports/yasm/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/yasm/portfile.cmake:32]: These tools use dynamic loading of `loadlibrary`, so dumpbin cannot analyze dependencies.

Comment (BillyONeal) [reply ports/yasm/portfile.cmake:32]: Just to confirm, you're saying "under normal circumstances, yes, but the thing on line 34 isn't a direct dependency already"?

Comment (BillyONeal) [reply ports/yasm/portfile.cmake:32]: Ah, no `AUTO_CLEAN` only cares about the executables, not their DLL dependencies.

**Review by BillyONeal — APPROVED** [review]

File: ports/yasm/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/yasm/portfile.cmake:32]: Yes, `AUTO_CLEAN` will only clean the folder when it only contains the executables and pdbs.

**Pull request merged**